operator/internal/controller/pulumi/stack_controller.go

```diff
@@ -756,22 +756,16 @@ func (r *StackReconciler) Reconcile(ctx context.Context, request ctrl.Request) (
 
 		requeueAfter := time.Duration(0)
 
+		// Try again with exponential backoff if the update failed.
 		if instance.Status.LastUpdate.State == shared.FailedStackStateMessage {
 			requeueAfter = max(1*time.Second, time.Until(instance.Status.LastUpdate.LastResyncTime.Add(cooldown(instance))))
 		}
-		if sess.stack.ContinueResyncOnCommitMatch {
+		// Schedule another poll if ContinueResyncOnCommitMatch is set or if we're tracking a branch.
+		if sess.stack.ContinueResyncOnCommitMatch || stack.Branch != "" {
 			requeueAfter = max(1*time.Second, time.Until(instance.Status.LastUpdate.LastResyncTime.Add(resyncFreq(instance))))
 		}
 		if stack.GitSource != nil {
-			trackBranch := len(stack.GitSource.Branch) > 0
-			if trackBranch {
-				// Reconcile every resyncFreq to check for new commits to the branch.
-				pollFreq := resyncFreq(instance)
-				log.Info("Commit hash unchanged. Will poll for new commits.", "pollFrequency", pollFreq)
-				requeueAfter = min(requeueAfter, pollFreq)
```

Comment:
`requeueAfter` defaults to 0 so this `min` was always disabling the requeue.

Reply:
Good find, that's totally the problem.

Regarding the duplication of code here, I think my rationale is that the resync frequency is unduly be coupled between branch tracking and the periodic resync, ie. the `resyncFrequencySeconds`. I anticipate new git-specific fields for branch tracking, such that one could say "check the branch every minute, and do a re-sync (for drift detection) every hour".

Reply:
I took the liberty of reverting the optimization, for clarity.